<img src="images/Logo_Tec_azul.png" width="500">

<center>

## Obtención de estadisticas descriptivas

### Herramientas computacionales: el arte de la analítica
        
### Oscar E. Delgadillo Ochoa - A01705935

### Pedro Oscar Pérez Murueta
    
### 15/09/2021


In [3]:
import pandas as pd
from matplotlib import pyplot as plt

In [4]:
df = pd.read_csv('dataset/Fortnite_players_stats.csv', index_col = 0)
df

Solo score  Solo top1  Solo kd  Solo winRatio  Solo matches  \
Player                                                                          
Prospеring           2476763       1828     4.37           18.0         10150   
BH nixxxay            439562       1694     8.71           29.1          5817   
Raпger               4519465       4582     9.60           34.0         13488   
Idk_Pi                752869        121     0.84            3.3          3687   
Twitch.GryphonRB     1136282       1327     6.32           30.0          4429   
...                      ...        ...      ...            ...           ...   
im bloom              106294         26     2.03            3.5           736   
Twitch kaOzs_         803918        396     3.42           13.2          3001   
slxyher               603939        310     1.69            5.1          6035   
RD-Antony             747295         71     1.10            1.9          3763   
O-HO-HO-HOǃ          1378689        161     1.68            2.3          7105   

                  Solo kills  Solo minutesPlayed  Duos score  Duos top1  \
Player                                                                    
Prospеring             36328               81389     4702684       5584   
BH nixxxay             35895               12732     4065613       4141   
Raпger                 85481              122171     1137279       1390   
Idk_Pi                  3005               32453     5326457       2249   
Twitch.GryphonRB       19591               36245     3692623       5743   
...                      ...                 ...         ...        ...   
im bloom                1440                4470     2208424       1551   
Twitch kaOzs_           8911               27375      890900        739   
slxyher                 9679               20545      432952        442   
RD-Antony               4048               31147     1010165        851   
O-HO-HO-HOǃ            11661               57572      305715        128   

                  Duos kd  ...  Squads kills  Squads minutesPlayed  LTM score  \
Player                     ...                                                  
Prospеring          10.71  ...       75787.0              117967.0   101589.0   
BH nixxxay           9.49  ...      117327.0              150439.0   151358.0   
Raпger               7.78  ...       76065.0               86684.0   452136.0   
Idk_Pi               1.02  ...        6892.0               80016.0  1748407.0   
Twitch.GryphonRB    11.13  ...       46068.0               81424.0   556721.0   
...                   ...  ...           ...                   ...        ...   
im bloom             2.53  ...       10842.0               25374.0    21840.0   
Twitch kaOzs_        2.88  ...       12628.0               38133.0   103451.0   
slxyher              1.90  ...       25626.0               67827.0   449563.0   
RD-Antony            5.23  ...       15107.0               50447.0   383709.0   
O-HO-HO-HOǃ          2.82  ...       12259.0               45511.0   331157.0   

                  LTM top1  LTM top3  LTM kd  LTM winRatio  LTM matches  \
Player                                                                    
Prospеring           170.0     118.0    5.78          26.8        634.0   
BH nixxxay           229.0     190.0    8.20          18.2       1259.0   
Raпger              1180.0     818.0   11.12          32.9       3583.0   
Idk_Pi               190.0     246.0    2.33           5.4       3539.0   
Twitch.GryphonRB     824.0     600.0   14.99          53.3       1546.0   
...                    ...       ...     ...           ...          ...   
im bloom               9.0      10.0    3.27           5.3        171.0   
Twitch kaOzs_         75.0      78.0    5.37          13.3        562.0   
slxyher              154.0     141.0    4.62           6.2       2497.0   
RD-Antony              8.0      10.0    4.28           0.6       1323.0   
O-HO-HO-HOǃ           16.0      20

In [5]:
new_df = df[['Solo minutesPlayed','Duos minutesPlayed', 'Trios minutesPlayed','Squads minutesPlayed', 'LTM minutesPlayed']]
new_df.describe()

Solo minutesPlayed  Duos minutesPlayed  Trios minutesPlayed  \
count         1435.000000         1435.000000          1411.000000   
mean         43517.135889        52877.229965         13274.673281   
std          36314.877771        38474.070183         10494.190693   
min            210.000000           20.000000             2.000000   
25%          16922.000000        21853.000000          5268.000000   
50%          35762.000000        43505.000000         10555.000000   
75%          59471.000000        75889.000000         19455.000000   
max         283835.000000       270990.000000         63145.000000   

       Squads minutesPlayed  LTM minutesPlayed  
count           1434.000000        1434.000000  
mean           40738.556485        8503.480474  
std            19507.459127        9849.725206  
min               74.000000          13.000000  
25%            29046.000000        2421.250000  
50%            39392.000000        5193.500000  
75%            51403.250000       11155.250000  
max           150439.000000      120381.000000

Al realizar el analísis de los datos sobre el videojuego Fortnite, decidí tomar como base el tiempo que emplean los 1435 en los diferentes modos de juego (solitario, duos, trios, escuadrones y ltm por sus siglas en inglés limited time modes), al comparar los diferentes tiempos, obteniendo la siguiente información:

+ El promedio de jugadores invierte jugando más tiempo (en minutos) en el modo Duos, tomando el siguiente orden:
    1. Duos - 52877.229965
    2. Solo - 43517.135889
    3. Squads - 40738.556485
    4. Trios - 13274.673281
    5. LTM (Limited Time Modes) - 8503.480474
   
+ El modo de juego que recibio un el mayor máximo en cuanto a minutos de jugadas fue el de Solo, tomando el siguiente orden:
    1. Solo - 283835.000000
    2. Duos - 270990.000000
    3. Squads - 150439.000000
    4. LTM (Limited Time Modes) - 120381.000000
    5. Trios - 63145.000000
    
+ El modo de juego que recibio un el mínimo de tiempo de minutos jugados fue el de Trios, tomando el orden siguiente:
    1. Trios - 2.000000
    2. LTM (Limited Time Modes) - 13.000000
    3. Duos - 20.000000
    4. Squads - 74.000000
    5. Solo - 210.000000
    
+ 

De los datos anteriores puntos pude llegar a la conclusión de que la mayoria de los jugadores que son casuales, sin embargo, hay un gran porcentaje que prefiere jugar en solitario, además de que los modos de juego que menor se usan son el Trio y LTM por sus siglas en inglés Limited Time Modes.


## Author
> #### Iyad Mohamed Fouad Ali

### Context
I'm a Junior Data Scientist and a Fortnite player, I wanted to make a fun project about fortnite, but I didn't find any dataset on kaggle, so I told my self why don't I make one.

### Content
In this dataset there is 5 game modes (Solo, Duo, Trios. Squads and ltm) each have:
Score: total score.
top1: total wins.
kd : kill/death ratio.
WinRatio: win ratio.
matches: number of matches played.
kills: number of eliminations made.
minutesPlayed: total number of minutes spent.

Data was collected during season 6 chapter 2 (4/2021)

### Acknowledgements
I collected the data using the fortnite tracker API from the tracker network website (https://tracker.gg/).